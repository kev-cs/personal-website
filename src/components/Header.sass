#header
  padding: 17.5rem 0 5rem
  text-align: center

  > div
    margin-bottom: 3rem

    #name
      font-size: $font-size-xxlarge
      font-weight: 500
      margin-bottom: 1rem
      color: $palette-primary-4

    #profession
      font-size: $font-size-large
      color: $palette-primary-4

#external-links
  display: flex
  margin: 3rem auto
  justify-content: space-evenly
  max-width: (640rem / 16)
  color: $palette-primary-0

  a
    flex-grow: 1
    flex-basis: 0
  i
    font-size: $font-size-xlarge

