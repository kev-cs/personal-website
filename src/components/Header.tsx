import * as React from "react";
import styles from "./Header.sass";
import { defineMessages, FormattedMessage } from "react-intl";

const messages = defineMessages({
  profession: {
    id: "profession",
    defaultMessage: "Software developer"
  },
  home: {
    id: "home",
    defaultMessage: "Home"
  },
  about: {
    id: "about",
    defaultMessage: "About"
  },
  connect: {
    id: "connect",
    defaultMessage: "Connect"
  }
});

export default class Header extends React.Component<{}, {}> {
  render(): JSX.Element {
    return (
      <header id={styles.header}>
        <div>
          <h1 id={styles.name}>Kevin Caro Silva</h1>
          <h2 id={styles.profession}>
            <FormattedMessage {...messages.profession}/>
          </h2>
        </div>
        <div id={styles.externalLinks}>
          <a>
            <i className="fab fa-github"/>
          </a>
          <a>
            <i className="fab fa-linkedin"/>
          </a>
          <a>
            <i className="fas fa-envelope"/>
          </a>
          <a>
            <i className="fas fa-comment-alt"/>
          </a>
        </div>
      </header>
    );
  }
}
