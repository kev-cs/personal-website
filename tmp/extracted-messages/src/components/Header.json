[
  {
    "id": "profession",
    "defaultMessage": "Software developer"
  },
  {
    "id": "home",
    "defaultMessage": "Home"
  },
  {
    "id": "about",
    "defaultMessage": "About"
  },
  {
    "id": "connect",
    "defaultMessage": "Connect"
  }
]