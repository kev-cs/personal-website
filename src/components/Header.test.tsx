import React from "react";
import Header from "./Header";
import renderer from "react-test-renderer";

import { IntlProvider } from "react-intl";

describe("Header", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <IntlProvider locale="en">{<Header/>}</IntlProvider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
