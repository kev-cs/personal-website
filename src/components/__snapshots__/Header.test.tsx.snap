// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  id="header"
>
  <div>
    <h1
      id="name"
    >
      Kevin Caro Silva
    </h1>
    <h2
      id="profession"
    >
      Software developer
    </h2>
  </div>
  <div
    id="externalLinks"
  >
    <a>
      <i
        className="fab fa-github"
      />
    </a>
    <a>
      <i
        className="fab fa-linkedin"
      />
    </a>
    <a>
      <i
        className="fas fa-envelope"
      />
    </a>
    <a>
      <i
        className="fas fa-comment-alt"
      />
    </a>
  </div>
</header>
`;
